### PR TITLE
drivers/hih6310: Initial commit of HIH6310 driver

### PR DESCRIPTION
**Pull request by gebart**
Honeywell HumidIcon Digital Humidity/Temperature Sensors: HIH-6130/6131 Series

Only basic humidity and temperature measurement support is implemented.

Missing:
 - Alarm interrupts
 - Command mode (reconfigure i2c address, alarm levels, alarm polarity, custom ID)

Where to buy: 
 - https://www.sparkfun.com/products/11295

### CODE REVIEW

File: Makefile.dep [hunk 1]
```diff
@@ -325,3 +325,8 @@ ifneq (,$(filter cpp11-compat,$(USEMODULE)))
   USEMODULE += timex
   FEATURES_REQUIRED += cpp
 endif
+
+ifneq (,$(filter hih6130,$(USEMODULE)))
+  USEMODULE += vtimer
+  FEATURES_REQUIRED += periph_i2c
```
Comment (gebart) [inline Makefile.dep]: Is this the correct place for adding the driver dependencies? (hih6130 uses vtimer internally)
Is `FEATURES_REQUIRED` correct to put here?

**Comment by PeterKietzmann** [comment]
@gebart I thinik it will take some more days to review this PR.

File: drivers/hih6130/hih6130.c [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+     * cycle, or data fetched before the first measurement has been completed.
+     */
+    HIH6130_STATUS_STALE_DATA   = 0x40,
+    HIH6130_STATUS_COMMAND_MODE = 0x80,
+    HIH6130_STATUS_DIAGNOSTIC   = 0xc0,
+};
+
+/** @brief Delay between requesting a measurement and data becoming ready */
+static const timex_t measurement_delay = {
+        .seconds = 0, .microseconds = 50 * MS_IN_USEC, };
```
Comment (PeterKietzmann) [inline drivers/hih6130/hih6130.c]: So you want a delay of 50000 us here? I just stumbled upon the `MS_IN_USEC` macro here.

File: drivers/hih6130/hih6130.c [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+    vtimer_sleep(measurement_delay);
+
+    status = hih6130_get_humidity_raw(dev, &tmp);
+
+    if (status < 0) {
+        return -1;
+    }
+
+    *relative_humidity_percent = tmp * (100.f / 16383.f);
```
Comment (PeterKietzmann) [inline drivers/hih6130/hih6130.c]: The data words are just 14 Bit long?

File: drivers/hih6130/hih6130.c [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    vtimer_sleep(measurement_delay);
+
+    status = hih6130_get_humidity_temperature_raw(dev, &hum_raw, &temp_raw);
+
+    if (status < 0) {
+        return -1;
+    }
+
+    *relative_humidity_percent = hum_raw * (100.f / 16383.f);
+    *temperature_celsius = temp_raw * (165.f / 16383.f) - 40.f;
```
Comment (PeterKietzmann) [inline drivers/hih6130/hih6130.c]: Where does the `165.f` come from?

File: drivers/hih6130/hih6130.c [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    return 0;
+}
+
+int hih6130_init(hih6130_t *dev, i2c_t i2c, uint8_t address)
```
Comment (PeterKietzmann) [inline drivers/hih6130/hih6130.c]: Could be also a `void` function

File: drivers/include/hih6130.h [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * @param[in]  dev                       Sensor device descriptor
+ * @param[out] relative_humidity_percent Measured relative humidity in percent
+ *
+ * @return 0 on success
+ * @return -1 on error
+ * @return 1 if data is stale
+ */
+int hih6130_get_humidity_float(hih6130_t *dev, float *relative_humidity_percent);
+int hih6130_get_humidity_temperature_float(hih6130_t *dev,
```
Comment (PeterKietzmann) [inline drivers/include/hih6130.h]: No documentation for that function? I would have assumed to have (a) just this one function that reads both values or (b) also to have a separate function to get the temperature value.

File: Makefile.dep (hunk 1 above)
Comment (PeterKietzmann) [inline Makefile.dep]: Honestly I don't know. I think most sensors define `FEATURES_REQUIRED` just in the makefile of their application. But doesn't mean that is the best solution. Are there any opinions by others?

File: drivers/include/hih6130.h (hunk 6 above)
Comment (gebart) [inline drivers/include/hih6130.h]: After thinking about it, I think the best way would be to remove the first
function and only keep the one which gets both values, with an added if
NULL check for the destination pointers.

Comment (PeterKietzmann) [inline drivers/include/hih6130.h]: Fine with me

File: drivers/hih6130/hih6130.c (hunk 2 above)
Comment (gebart) [inline drivers/hih6130/hih6130.c]: yes. The number 50 is arbitrary, there is some conflicting information regarding the measurement time of the sensor in the vendor documentation (one place says 37.5 ms, somewhere else I read maximum 100 ms, some other much smaller number nominally) but 50 ms seem to work.

**Comment by gebart** [comment]
addressed comments

**Comment by PeterKietzmann** [comment]
Does anyone own this sensor and is able to test this PR?!

**Comment by gebart** [comment]
OK to squash?

**Comment by PeterKietzmann** [comment]
Yes. Who owns this hardware?

**Comment by gebart** [comment]
squashed

I own the hardware but I assume we are looking for anyone who did not write the code to test it.

File: drivers/hih6130/hih6130.c (hunk 4 above)
Comment (gebart) [inline drivers/hih6130/hih6130.c]: the formula is taken directly from the data sheet. the 165 is the span of the sensor (-40 .. +125 C), the -40 is the offset

File: drivers/hih6130/hih6130.c (hunk 3 above)
Comment (gebart) [inline drivers/hih6130/hih6130.c]: yes, the topmost two bits are status bits and are filtered out in the _raw function

**Comment by PeterKietzmann** [comment]
needs rebase

**Comment by gebart** [comment]
rebased

File: Makefile.dep (hunk 1 above)
Comment (PeterKietzmann) [inline Makefile.dep]: Ahh, I forgot this one, sorry. With #3420 it is surely the correct place for the dependencies. I tend to take the `FEATURES_REQUIRED` out here. I think we used to put it in an applications makefile to throw a warning if features don't match (as you already did).

Comment (gebart) [inline Makefile.dep]: OK will update

**Comment by gebart** [comment]
OK to squash?

**Comment by gebart** [comment]
removed FEATURES_REQUIRED and squashed

**Comment by PeterKietzmann** [comment]
What's up with Travis again? I didn't follow the latest CI related PRs...

**Comment by gebart** [comment]
coap is broken because sf.net has been having problems since yesterday.

**Comment by PeterKietzmann** [comment]
@gebart, rebase should make Travis happy again (after #3454 was merged). I'd merge this PR then.

**Comment by gebart** [comment]
Failures are all coap (because of broken mirror) and libfixmath_unittests (race condition in makefile)

**Comment by OlegHahm** [comment]
Let's merge #3460 quickly to get rid of at least one of these issues in Travis.

**Pull request merged**